game/entity/enemy.py

```diff
@@ -76,13 +79,49 @@ def create_full_path(self):
             ):
                 self.path.append(center_of_tile(*point))
 
-    def move_along_path(self):
-        self.pathidx += 1  # TODO: Handle direction facing
-        self.pathidx %= len(self.path)
-        self.position = self.path[self.pathidx]
+    def move_one(self):
+        """
+        Moves the enemy forward one unit along his path
+        """
+        if self.movesleft > 0:
+            self.pathidx += 1
+            self.pathidx %= len(self.path)
+            self.position = self.path[self.pathidx]
+            self.movesleft -= 1
+            self.update_vision()
+
+    def update_direction(self):
+        """
+        Changes the direction the enemy is facing to face the next path location
+        """
+        nextposition = self.path[(self.pathidx + 1) % len(self.path)]
+        basedirection = nextposition - self.position
+        if basedirection[1] > 0:
+            self.direction = Direction.NORTH
+        elif basedirection[1] < 0:
+            self.direction = Direction.SOUTH
+        elif basedirection[0] > 0:
+            self.direction = Direction.EAST
+        elif basedirection[0] < 0:
+            self.direction = Direction.WEST
+        self.update_vision()
+
+    def update_vision(self):
+        self.vision_points = []
+        for visiondistance in range(1, self.maxvision + 1):
+            visionpoint = self.position + self.direction * visiondistance * TILE_SIZE
+            if arcade.get_sprites_at_exact_point(visionpoint, self._walls):
+                break
+            self.vision_points.append(visionpoint)
 
     def update(self):
-        self.move_along_path()
+        self.movesleft = self.movecount
 
     def draw_path(self):
         arcade.draw_line_strip(self.path + self.path[:1], self.pathcolor, 2)
+
+    def draw_vision(self):
+        for vision_point in self.vision_points:
+            arcade.draw_circle_filled(
+                vision_point[0], vision_point[1], 5, arcade.color.WISTERIA, 3
```

Comment:
have you stress tested this? i'm fairly certain this will be quite slow once we start increasing the amount of guards. for now though this should be ok...

Reply:
This wasn't meant to be permanent.  I'm not sure how we want to render the vision and what we want it to look like, but this probably isn't it.  The game engine handles moving sprites around with more efficiency, right? But as I said in my original commit, maybe this makes more sense as a dimming of everywhere except where the guards have vision, though that might be even less efficient.  Either way the test map has 7 guards you can test this with.

Reply:
Maybe I'll just make them transparent bright yellow square sprites in a sprite list in order of how far they are from a guard.  I will have to figure out what to do with the extra sprites when their vision hits a wall.